### ISSUE TITLE
high memory usage/crash

### ISSUE THREAD

**Issue by Spongman** [codeout/vscode-junos#1]
when i start VScode with this extension (and _no_ others) installed, one of the vscode sub-processes (probably the extension host) quickly uses up all available memory and causes the whole machine to grind to a halt before the OS kills it.

![image](https://user-images.githubusercontent.com/1088194/52148535-dc652000-261e-11e9-8045-ce684aab8850.png)

**Comment by codeout** [comment]
Thanks for your report, it mostly sounds like a bug, the process might be stuck in the loop or just a memory leak.

Is there any chance I can try files you opened?
I couldn't reproduce the problem on my VM environment.

![image](https://user-images.githubusercontent.com/921696/52166458-d66f5c00-2750-11e9-879c-a585b7090d78.png)

windows: 10 Pro
vscode ver: 1.30.2
RAM: 6GB

**Comment by Spongman** [comment]
here's the minimal repro:

```
system {
    license {
        autoupdate {
            url https://ae1.juniper.net/junos/key_retrieval;
        }
    }
}
```

be careful, you'll need to power-cycle if you don't kill it in time.

**Comment by codeout** [comment]
I couldn't reproduce with the config above - vscode process is alive for 5 minutes.
`system { ... }` style configuration is not supported for now actually, it's still under development. (Only `set system license ...` style is supported today.) But we need some protection to avoid system crash meanwhile.

![image](https://user-images.githubusercontent.com/921696/52434868-be824880-2b53-11e9-9578-95e8fcee5a87.png)

I'll take a look into the process behavior soon.

**Comment by codeout** [comment]
@Spongman If you still see the problem, could you try the latest version, 0.1.4?
A similar problem was fixed at #2.
